clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -7147,7 +7147,9 @@ void Sema::ProcessDeclAttributeList(
   // good to have a way to specify "these attributes must appear as a group",
   // for these. Additionally, it would be good to have a way to specify "these
   // attribute must never appear as a group" for attributes like cold and hot.
-  if (!D->hasAttr<OpenCLKernelAttr>()) {
```

Comment:
This part checks OpenCL attributes. The condition should be kept unchanged.

Reply:
It doesn't check OpenCL attributes, it checks everything that's not an OpenCL kernel. That's why this currently causes errors. If you want I could put `&& LangOpt.IsOpenCL` but that causes other tests to fail so this was the less invasive option.

Reply:
Further down, we check explicitly for `CUDAGlobalAttr`, which implies this is intended to be checked against non-OpenCL code.